graphql-server/src/utils/oncotree.ts

```diff
@@ -110,3 +112,50 @@ async function getOncotreeCodesFromNeo4j() {
     await session.close();
   }
 }
+
+/**
+ * Include cancer type fields in the search query indirectly by searching the Oncotree cache
+ * for matching cancerType or cancerTypeDetailed values. If a match is found, add the corresponding
+ * Oncotree code to the original search query.
+ */
+export function includeCancerTypeFieldsInSearch(
+  where: SampleWhere,
+  oncotreeCache: NodeCache
+) {
+  const customWhere = { ...where };
+
+  const sampleMetadataFilters =
+    customWhere?.hasMetadataSampleMetadata_SOME?.OR ||
+    customWhere?.OR?.find((filter) => filter.hasMetadataSampleMetadata_SOME)
+      ?.hasMetadataSampleMetadata_SOME?.OR;
+
+  if (sampleMetadataFilters?.length) {
+    const searchInput = Object.values(sampleMetadataFilters[0])[0] as
+      | string
+      | string[];
+    const searchValues = Array.isArray(searchInput)
+      ? searchInput
+      : [searchInput];
+
+    const addOncotreeCode = (code: string) => {
+      sampleMetadataFilters?.push({ oncotreeCode_IN: [code] });
+    };
+
+    oncotreeCache.keys().forEach((code) => {
+      const { name, mainType } = (oncotreeCache.get(
+        code
+      ) as CachedOncotreeData)!;
+      if (
+        searchValues.some(
+          (val) =>
+            name?.toLowerCase().includes(val.toLowerCase()) ||
+            mainType?.toLowerCase().includes(val.toLowerCase())
+        )
+      ) {
+        addOncotreeCode(code);
+      }
+    });
```

Comment:
Iterate through each Oncotree code in the cache. For each code, look at its respective `name` and `mainType` values. If either partially matches the user input, we'll add the corresponding Oncotree code to the `where`/filter that will be included in the database query.